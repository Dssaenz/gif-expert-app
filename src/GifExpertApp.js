import React, { useState } from 'react'

import AddCategory from './components/add-category'
import GifGrid from './components/gif-grid'

const GifExpertApp = ({ defaultCategories = [] }) => {
	const [categories, setCategories] = useState(defaultCategories);

	return (
		<>
			<h2>gif expert app</h2>
			<br />
			<AddCategory setCategories={setCategories} />
			{categories.map(item => (
				<GifGrid
					key={item}
					categories={item}
				/>
			))}
		</>
	)
}

export default GifExpertApp