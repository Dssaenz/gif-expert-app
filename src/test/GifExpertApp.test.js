import React from 'react'
// Dependencies
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';


describe('Test in <GifExpertApp /> component', () => {

	test('It should display the component correctly', () => {

		const categories = ['Hola', 'Mundo'];
		const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find('GifGrid').length).toBe(categories.length)
		expect(wrapper.find('AddCategory')).toEqual({});
	})
})