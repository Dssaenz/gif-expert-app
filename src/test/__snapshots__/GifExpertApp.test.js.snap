// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifExpertApp /> component It should display the component correctly 1`] = `
<Fragment>
  <h2>
    gif expert app
  </h2>
  <br />
  <AddCategory
    setCategories={[Function]}
  />
  <GifGrid
    categories="Hola"
    key="Hola"
  />
  <GifGrid
    categories="Mundo"
    key="Mundo"
  />
</Fragment>
`;
